### PR TITLE
add [functional] attribute to gas calculating functions

### PR DESCRIPTION
**Pull request by daejunpark**
These C* and R* functions are total.

### CODE REVIEW

**Comment by daejunpark** [comment]
For the second commit, see the commit message, which is copied below:

--

Indeed, the following two rule names does not exist:

    EVM.#lookup.none
    EVM.#lookup.some

They should have been:

    EVM-TYPES.#lookup.none
    EVM-TYPES.#lookup.some

This is a bug introduced at: 223f21c#diff-558a5f8e4337c04713ebea82c99dde22R19-R20

File: tests/specs/concrete-rules.txt [hunk 1]
```diff
@@ -5,8 +5,6 @@ EVM.Cgascap
 EVM.Cmem
 EVM.Csstore.new
 EVM.Csstore.old
-EVM.#lookup.none
```
Comment (denis-bogdanas) [inline tests/specs/concrete-rules.txt:8]: This is probably related to my PR. Thanks for pointing out: https://github.com/kframework/evm-semantics/pull/852 If they don't cause any test failures let me fix this part pls.

File: evm.md [hunk 2]
```diff
@@ -2070,12 +2070,12 @@ There are several helpers for calculating gas (most of them also specified in th
     rule <k> Cselfdestruct(SCHED, ISEMPTY:Bool, BAL)
           => Gselfdestruct < SCHED > +Int Cnew(SCHED, ISEMPTY andBool Gselfdestructnewaccount << SCHED >>, BAL) ... </k>
 
-    syntax Int ::= Cgascap ( Schedule , Int , Int , Int ) [function]
-                 | Csstore ( Schedule , Int , Int , Int ) [function]
-                 | Rsstore ( Schedule , Int , Int , Int ) [function]
-                 | Cextra  ( Schedule , Bool , Int )      [function]
-                 | Cnew    ( Schedule , Bool , Int )      [function]
-                 | Cxfer   ( Schedule , Int )             [function]
+    syntax Int ::= Cgascap ( Schedule , Int , Int , Int ) [function, functional]
```
Comment (ehildenb) [inline evm.md:2073]: This uses `#allBut64th` which is not marked `functional`, please either mark dependencies as `functional` or don't mark this one as functional.

Comment (daejunpark) [reply evm.md:2073]: done

**Comment by daejunpark** [comment]
@denis-bogdanas @ehildenb I dropped the concrete-rules commit.  I found that there are many concrete-rules.txt files that have the same bug.  I'll make another PR that fixes all of them.

File: tests/specs/concrete-rules.txt (hunk 1 above)
Comment (daejunpark) [reply tests/specs/concrete-rules.txt]: @denis-bogdanas I made this as a separate PR: https://github.com/kframework/evm-semantics/pull/861

**Pull request merged**